Mathlib/Topology/Metrizable/CompletelyMetrizable.lean

```diff
@@ -36,6 +43,108 @@ variable {X Y : Type*}
 
 namespace TopologicalSpace
 
+/-- A topological space is completely pseudometrizable if there exists a pseudometric space
+structure compatible with the topology which makes the space complete.
+To endow such a space with a compatible distance, use
+`letI := upgradeIsCompletelyPseudoMetrizable X`. -/
+class IsCompletelyPseudoMetrizableSpace (X : Type*) [t : TopologicalSpace X] : Prop where
+  complete : ∃ m : PseudoMetricSpace X, m.toUniformSpace.toTopologicalSpace = t ∧
+    @CompleteSpace X m.toUniformSpace
+
+instance (priority := 100) _root_.PseudoMetricSpace.toIsCompletelPseudoMetrizableSpace
+    [PseudoMetricSpace X] [CompleteSpace X] : IsCompletelyPseudoMetrizableSpace X :=
+  ⟨⟨‹_›, rfl, ‹_›⟩⟩
+
+/-- A convenience class, for a completely pseudometrizable space endowed with a complete
+pseudometric. No instance of this class should be registered: It should be used as
+`letI := upgradeIsCompletelyPseudoMetrizable X` to endow a completely pseudometrizable
+space with a complete pseudometric. -/
+class UpgradedIsCompletelyPseudoMetrizableSpace (X : Type*) extends
+  PseudoMetricSpace X, CompleteSpace X
+
+open scoped Uniformity in
+instance (priority := 100) IsCompletelyPseudoMetrizableSpace.of_completeSpace_pseudometrizable
+    [UniformSpace X] [CompleteSpace X] [(𝓤 X).IsCountablyGenerated] :
+    IsCompletelyPseudoMetrizableSpace X where
+  complete := ⟨UniformSpace.pseudoMetricSpace X, rfl, ‹_›⟩
+
+/-- Construct on a completely pseudometrizable space a pseudometric (compatible with the topology)
+which is complete. -/
+noncomputable def completelyPseudoMetrizableMetric (X : Type*) [TopologicalSpace X]
+    [h : IsCompletelyPseudoMetrizableSpace X] : PseudoMetricSpace X :=
+  h.complete.choose.replaceTopology h.complete.choose_spec.1.symm
+
+theorem complete_completelyPseudoMetrizableMetric (X : Type*) [ht : TopologicalSpace X]
+    [h : IsCompletelyPseudoMetrizableSpace X] :
+    @CompleteSpace X (completelyPseudoMetrizableMetric X).toUniformSpace := by
+  convert h.complete.choose_spec.2
+  exact PseudoMetricSpace.replaceTopology_eq _ _
+
+/-- This definition endows a completely pseudometrizable space with a complete pseudometric.
+Use it as: `letI := upgradeIsCompletelyPseudoMetrizable X`. -/
+noncomputable
+def upgradeIsCompletelyPseudoMetrizable (X : Type*) [TopologicalSpace X]
+    [IsCompletelyPseudoMetrizableSpace X] :
+    UpgradedIsCompletelyPseudoMetrizableSpace X :=
+  letI := completelyPseudoMetrizableMetric X
+  { complete_completelyPseudoMetrizableMetric X with }
+
+namespace IsCompletelyPseudoMetrizableSpace
+
+/-- Note: the priority is set to 90 to ensure that this instance is only applied after
+`PseudoEMetricSpace.pseudoMetrizableSpace`. This prevents unnecessary attempts to infer
+completeness. -/
+instance (priority := 90) PseudoMetrizableSpace [TopologicalSpace X]
+    [IsCompletelyPseudoMetrizableSpace X] : PseudoMetrizableSpace X := by
```

Comment:
I almost wonder whether this instance should be scoped. In failing searches for `PseudoMetrizableSpace` Lean will still have to search for `CompleteSpace`.

If we bench this and everything is fine, I will withdraw my concern.